ratis-server/src/main/java/org/apache/ratis/server/leader/InstallSnapshotRequests.java

```diff
@@ -143,11 +143,9 @@ public InstallSnapshotRequestProto next() {
     }
 
     private InstallSnapshotRequestProto newInstallSnapshotRequest(FileChunkProto chunk, boolean done) {
-      synchronized (server) {
-        final SnapshotChunkProto.Builder b = LeaderProtoUtils.toSnapshotChunkProtoBuilder(
-            requestId, requestIndex++, snapshot.getTermIndex(), chunk, totalSize, done);
-        return LeaderProtoUtils.toInstallSnapshotRequestProto(server, followerId, b);
-      }
+      final SnapshotChunkProto.Builder b = LeaderProtoUtils.toSnapshotChunkProtoBuilder(
+          requestId, requestIndex++, snapshot.getTermIndex(), chunk, totalSize, done);
+      return LeaderProtoUtils.toInstallSnapshotRequestProto(server, followerId, b);
```

Comment:
Let's keep `synchronized (server)`.  Otherwise, it could create an inconsistent request.

Reply:
Thanks for the review, I will make changes